server/resp/src/main/java/org/infinispan/server/resp/Resp3Handler.java

```diff
@@ -76,10 +81,7 @@ public CompletionStage<RespRequestHandler> delegate(ChannelHandlerContext ctx,
       return super.actualHandleRequest(ctx, command, arguments);
    }
 
-   public void checkPermission(AuthorizationPermission authorizationPermission) {
-      AuthorizationManager authorizationManager = cache.getAuthorizationManager();
-      if (authorizationManager != null) {
-         authorizationManager.checkPermission(authorizationPermission);
-      }
+   public void checkPermission(Subject subject, AuthorizationPermission authorizationPermission) {
+      SecurityActions.checkPermission(respServer.getCacheManager(), subject, authorizationPermission);
```

Comment:
This code is wrong. 
Some authz checks need to be done against the cache permissions, while others may need to be done using the global permissions.